build.gradle

```diff
@@ -1,19 +1,60 @@
 plugins {
     id 'java'
+    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
+    id 'org.springframework.boot' version '2.4.3'
+
 }
 
-group 'org.example'
-version '1.0-SNAPSHOT'
+subprojects {
+    apply plugin: 'org.springframework.boot'
+    apply plugin: 'io.spring.dependency-management'
+    apply plugin: 'java-library'
 
-repositories {
-    mavenCentral()
-}
+    group = 'com.epam.esm'
+    version = '1.0.0'
+    sourceCompatibility = 1.8
+    targetCompatibility = 1.8
 
-dependencies {
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.0'
-    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine'
-}
+    configurations {
+        compileOnly {
+            extendsFrom annotationProcessor
+        }
+    }
+
+    repositories {
+        mavenCentral()
+    }
+
+    dependencies {
+        implementation group: 'org.springframework.boot', name: 'spring-boot-starter-parent', version: '2.4.3', ext: 'pom'
+        implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
+        implementation 'org.springframework.boot:spring-boot-starter-hateoas'
+
+        implementation 'org.springframework.boot:spring-boot-starter'
+        implementation group: 'org.springframework.boot', name: 'spring-boot-starter-actuator', version: '2.4.3'
+        compile group: 'org.modelmapper', name: 'modelmapper', version: '2.3.9'
+        compile group: 'net.jodah', name: 'typetools', version: '0.6.2'
+        compile group: 'com.mchange', name: 'c3p0', version: '0.9.5.2'
+        compile group: 'org.hibernate', name: 'hibernate-envers', version: '5.4.27.Final'
+        compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.11'
+
+        implementation group: 'javax.validation', name: 'validation-api', version: '2.0.1.Final'
+        testCompile('org.springframework.boot:spring-boot-starter-test')
+        compileOnly 'org.projectlombok:lombok'
+        runtimeOnly 'mysql:mysql-connector-java'
+        annotationProcessor 'org.projectlombok:lombok'
+
+
+        testImplementation 'org.junit.jupiter:junit-jupiter-api:5.4.2'
+        testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.4.2'
+        testImplementation group: 'org.mockito', name: 'mockito-core', version: '3.8.0'
+        testImplementation group: 'org.mockito', name: 'mockito-junit-jupiter', version: '3.8.0'
```

Comment:
The compile and testCompile configurations inherited from the Java plugin are still available but are deprecated. You should avoid using them, as they are only kept for backwards compatibility.

Reply:
All compile and testCompile configurations were replaced by implimentations